### ISSUE TITLE
add ability to switch the lexer used in the syntax highlighting seperate from file extension

### ISSUE THREAD

**Issue by spotandjake** [replit-discord/replit-desktop#149]
Frissyn can you implement this it does not need to be ace compatible for the time being. this should be fairly easy to implement through the monaco api you found the other day. and you can add the area to modify it too the theme editor i added. along with this could be added to reflux scripts as well.

**Comment by frissyn** [comment]
Yeah, I've been experimenting with the monaco API that repl.it uses and I could probably get smth like this working. Problem is, the editor doesn't reload unless it's been changed by an action dispatched on the store, and I don't know what this action is.

It's on my to-do list @spotandjake

**Comment by spotandjake** [comment]
If you send me the code to change it i will try to get it too reload.

**Comment by cursorweb** [comment]
would you even be able to access the editor instance?

**Comment by frissyn** [comment]
> would you even be able to access the editor instance?

Yeah, we've already got examples that get the instance from the window: `window.monaco.editor`.

**Comment by spotandjake** [comment]
Yes alternatively I think we could replace the editor isntqnce with a custom instance or use a remap to a custom replenvironment script that could open it  #all.
